plugin.lua

```diff
@@ -71,13 +70,21 @@ local on_event = require("factorio-plugin.on-event")
 ---@param text string @ The content of file
 ---@return nil|Diff[]
 function OnSetText(uri, text)
-  if text:sub(1, 4) == "--##" then return end
+  ---I can't see a reason to process ---@meta files
+  ---Speeds up loading by not reading mod debugger annotations.
+  local sub = text:sub(1, 8)
+  if sub == "---@meta" or sub == "--##" then return end
```

Comment:
Ignore ---@meta files?  Thoughts? It improves load times when Factorio typings are in the library

Reply:
Pretty sure there once was something in those files that needed to be touched by the plugin but right now that is not the case. However jarg has been talking about a custom `---@event` tag which the plugin would have to see. But since that is in the future and that could be put in a separate file by the generator, sure I'm for it. Although when I wrote the typedef generator I added `--##` at the top, I wonder if/why jarg removed that. But regardless of if it was removed in the generated type defs, ignoring builltin `---@meta` files is good anyway.

Reply:
>However jarg has been talking about a custom ---@event tag which the plugin would have to see.

One of the things that my brain takes a second to process is `--##` Whats your thoughts on using something like
`---@plugin disable` and `---@plugin enable` to disable and enable blocks. It is more inline with with Sumneko styling and could be further expaned to `---@plugin disable: remote` to only disable it for a specific part of the plugin.

Reply:
The reason we started with `--##` was because that's the example (or was the example at least) in the `sumneko.lua` wiki for plugins. I've since kept it because it works and I didn't want to change it just to change it.
I've though about the enabling and disabling of the plugin or its modules for sections of a file but this isn't exactly easy to do with Lua patterns without sacraficing a notable amount of performance that scales with the amount of `---@plugin` tags in a file. It might not be super bad but I've never done it because I didn't see the need to. The cases where I saw a potential need to disable it I've added an "inline" `--##` to disable the plugin for that one case (like remote.add_interface). Plus my approach is that the plugin should be good enough to where you don't have the need to disable it which leads to fewer options for the user which makes it easier to use. And in my experience `--##` has only ever been used by myself to tell other people to test if the plugin is actually causing an issue, I've never heard or seen anything about it outside of that.
So TL;DR: I don't consider this to be an issue meaning increasing the complexity doesn't seem worth it